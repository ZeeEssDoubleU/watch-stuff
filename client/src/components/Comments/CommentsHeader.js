import React from "react";
import PropTypes from "prop-types";
import { Button, Icon } from "semantic-ui-react";

import "./CommentsHeader.scss";

const CommentsHeader = props => {
	const commentsCount = props.commentsCount
		? Number(props.commentsCount).toLocaleString()
		: props.commentsCount;
	return (
		<div className="comments-header">
			<h4>{commentsCount} Comments</h4>
			<Button
				className="sort-button"
				basic
				compact
				icon
				labelPosition="left">
				<Icon name="align left" />
				SORT BY
			</Button>
		</div>
	);
};

CommentsHeader.propTypes = {};

export default CommentsHeader;
