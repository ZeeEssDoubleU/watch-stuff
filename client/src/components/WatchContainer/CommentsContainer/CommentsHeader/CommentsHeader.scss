.comments-header {
   display: flex;
   align-items: center;
   margin-bottom: 10px;
   h4 {
      margin: 0 16px 0 0;
   }
}