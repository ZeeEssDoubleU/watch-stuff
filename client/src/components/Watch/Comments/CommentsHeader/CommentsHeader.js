import React from "react";
import PropTypes from "prop-types";

import "./CommentsHeader.scss";

const CommentsHeader = props => {
	return <div className="comments-header">Comment Header here.</div>;
};

CommentsHeader.propTypes = {};

export default CommentsHeader;
