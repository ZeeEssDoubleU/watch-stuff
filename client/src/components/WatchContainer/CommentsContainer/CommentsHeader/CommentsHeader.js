import React from "react";
import PropTypes from "prop-types";
import { Button, Icon } from "semantic-ui-react";

import "./CommentsHeader.scss";

const CommentsHeader = props => {
	return (
		<div className="comments-header">
			<h4>{props.commentCount} Comments</h4>
			<Button className="sort-button" basic compact icon labelPosition="left">
				<Icon name="align left" />
				SORT BY
			</Button>
		</div>
	);
};

CommentsHeader.propTypes = {};

export default CommentsHeader;
