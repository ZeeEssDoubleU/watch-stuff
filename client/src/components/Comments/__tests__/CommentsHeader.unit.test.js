import React from "react";
import { shallow } from "enzyme";

import CommentsHeader from "../CommentsHeader";

describe("CommentsHeader", () => {
	test("renders with props.commentCount = null", () => {
		const wrapper = shallow(<CommentsHeader />);
		expect(wrapper).toMatchSnapshot();
	});

	test("renders with props.commentCount = 0", () => {
		const wrapper = shallow(<CommentsHeader amountComments={123} />);
		expect(wrapper).toMatchSnapshot();
	});
});
