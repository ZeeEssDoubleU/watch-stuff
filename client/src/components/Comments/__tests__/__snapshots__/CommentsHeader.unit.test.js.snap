// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentsHeader renders with props.commentCount = 0 1`] = `
<div
  className="comments-header"
>
  <h4>
     Comments
  </h4>
  <Button
    as="button"
    basic={true}
    className="sort-button"
    compact={true}
    icon={true}
    labelPosition="left"
  >
    <Icon
      as="i"
      name="align left"
    />
    SORT BY
  </Button>
</div>
`;

exports[`CommentsHeader renders with props.commentCount = null 1`] = `
<div
  className="comments-header"
>
  <h4>
     Comments
  </h4>
  <Button
    as="button"
    basic={true}
    className="sort-button"
    compact={true}
    icon={true}
    labelPosition="left"
  >
    <Icon
      as="i"
      name="align left"
    />
    SORT BY
  </Button>
</div>
`;
